pkg/chrootarchive/archive.go

```diff
@@ -48,16 +46,20 @@ func untarHandler(tarArchive io.Reader, dest string, options *archive.TarOptions
 		}
 	}
 
-	r := ioutil.NopCloser(tarArchive)
+	r := tarArchive
```

Comment:
Did you remove the other section that commented out the defer close from the caller?

I really am not a fan of this approach of testing for `io.Closer` and closing if it is. The `http` package does this and it causes havoc trying to figure out why things are getting closed at inopportune times. I agree with fixing the race but it doesn't seem related to your crash nor the correct fix for it.